department-of-veterans-affairs/abd-vro
# BIP Integration: updates EP Merge App and adds end-2-end integration test

## What was the problem?

Before these changes, EP merge app would fail upon making some downstream calls to `svc-bip-api`. These changes allow it to work with with updates made to `svc-bip-api`, and the downstream `mock-bip-claims-api`

Associated tickets or Slack threads:
- #2091
- Blocked by:
  - #2195 
  - #2247 

## How does this fix it?[^1]

Changes step in ep-merge-machine where contentions were updated on pending claim to call the create contentions endpoint, not the update endpoint. 

Adds integration test that runs after current pytest integration tests to github action. New test starts `ee-ep-merge-app`, `svc-bip-api`, and `mock-bip-claims-api`. Then, a request is made to the `ee-ep-merge-app`merge endpoint to start a merge request. Then the endpoint to get the merge job status is queried and checked to see if the merge job was successful.  

## How to test this PR
- pull code
- from `/domain-ee/ee-ep-merge-app` run `pytest`
- from `/domain-ee/ee-ep-merge-app` run `pytest ./integration`
- run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
